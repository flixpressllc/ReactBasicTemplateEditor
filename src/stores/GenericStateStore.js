import { clone, traverseObject, toType } from 'happy-helpers';

/*
Allowed states will look something like this:

  allowedStates = {
    templateType: 'string',
    templateId: 'number',
    caughtErrors: 'array'
  }

  Starting values will look something like this:

    allowedStates = {
      templateType: '',
      caughtErrors: []
    }
*/

class GenericStateStore {
  constructor (allowedStates, startingValues) {
    this._setupAllowedStates(allowedStates);
    this._setupStartingValues(startingValues);
  }

  _setupAllowedStates (allowedStates) {
    if (toType(allowedStates) !== 'object') {
      throw new Error(`A generic state store cannot be instanciated without an 'allowedStates' object. ${allowedStates} given.`)
    }
    this.allowedStates = allowedStates;
  }

  _setupStartingValues (startingValues) {
    if (startingValues) {
      this.storedState = startingValues;
    }
  }

  getState (stringOrArrayOfKeys) {
    if (stringOrArrayOfKeys === undefined) return clone(this.storedState);
    if (toType(stringOrArrayOfKeys) === 'array') {
      return this._getStateByArrayOfKeys(stringOrArrayOfKeys);
    }
    if (toType(stringOrArrayOfKeys) === 'string') {
      return this._getStateByKey(stringOrArrayOfKeys);
    }
  }

  // Returns a value only
  _getStateByKey (key) {
    if (this.allowedStates[key] === undefined) {
      throw new Error(`The desired state property "${key}" from StateStore is not allowed.`);
    }
    if (['object', 'array'].indexOf(this.allowedStates[key]) > -1) {
      return clone(this.storedState[key]);
    }
    return this.storedState[key];
  }

  // Returns an object of key: value pairs
  _getStateByArrayOfKeys (arrayOfKeys) {
    let objects = arrayOfKeys.map(key => {
      return {[key]: this.getStateByKey(key)} // ES2015 computed property names
    });
    return Object.assign(...objects);
  }

  setState (obj) {
    this._storedStatesAreAllowed(obj);
    let newState = Object.assign({}, this.storedState, obj);
    if (!this._storedStatesAreEqual(newState)) {
      this.storedState = newState;
      return true;
    }
    return false;
  }

  _storedStatesAreAllowed (obj) {
    traverseObject(obj, (k, v) => {
      if (this.allowedStates[k] === undefined) {
        throw new Error(`There is no entry for "${k}" in allowed states.`);
      }
      if (this.allowedStates[k] !== toType(v)) {
        throw new Error(`Expected the type of ${k} to be ${this.allowedStates[k]}. ${toType(v)} was given.`)
      }
    })
  }

  _storedStateExists (key) {
    return this.storedState[key] !== undefined;
  }

  _storedStateValueEquals (key, newValue) {
    if (!this._storedStateExists(key)) return false;
    let stateValue = this.storedState[key];
    let theType = toType(stateValue);

    if (theType !== toType(newValue)) return false;
    if (['array','object'].indexOf(theType) > -1) {
      return JSON.stringify(newValue) === JSON.stringify(stateValue);
    }
    return newValue === stateValue;
  }

  _storedStatesAreEqual (newState) {
    let equal = true;
    traverseObject(newState, (k, v) => {
      if (!this._storedStateValueEquals(k, v)) {
        equal = false;
      }
    })
    return equal;
  }
}

export default GenericStateStore;
