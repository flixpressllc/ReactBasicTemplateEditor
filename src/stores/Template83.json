{
  "templateId": 83,
  "textFields": {
    "Top Line": {
      "value": "",
      "previewImage": "t83_1.jpg"
    },
    "Bottom Line": {
      "value": "",
      "previewImage": "t83_1.jpg"
    }
  },
  "ui": [
    {
      "Properties": [
        {
          "type": "TextField",
          "name": "Top Line"
        },
        {
          "type": "TextField",
          "name": "Bottom Line"
        }
      ]
    }
  ]
}