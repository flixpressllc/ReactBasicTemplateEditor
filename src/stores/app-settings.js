export const YOU_TUBE_API_KEY = 'AIzaSyBxurdcfDBk6hKm3l_vk3MWHGWle47RdMQ';
export const XML_CONTAINER_ID = 'RndTemplate_HF';
export const IMAGES_CONTAINER_ID = 'CroppedImageFilenames';

const LOCAL_DEV_THUMBNAIL_PATH = '/images/';
const PRODUCTION_THUMBNAIL_PATH = '/templates/TB/';

function thumbnailPath () {
  if (window.location.hostname === 'localhost') return LOCAL_DEV_THUMBNAIL_PATH;
  return PRODUCTION_THUMBNAIL_PATH;
}

export const THUMBNAIL_URL_PREFIX = thumbnailPath();
