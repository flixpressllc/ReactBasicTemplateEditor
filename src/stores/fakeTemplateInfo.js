var fakeTemplateInfo = {
  "83": {
    "templateId": 83,
    "textFields": {
      "Top Line": {
        "value": "",
        "previewImage": "t83_1.jpg"
      },
      "Bottom Line": {
        "value": "",
        "previewImage": "t83_1.jpg"
      }
    },
    "ui": [
      {
        "Properties": [
          {
            "type": "TextField",
            "name": "Top Line"
          },
          {
            "type": "TextField",
            "name": "Bottom Line"
          }
        ]
      }
    ]
  },
  "84": {
    "templateId": 84,
    "textFields": {
      "Top Line": {
        "value": "",
        "previewImage": "t84_1.jpg"
      },
      "Bottom Line": {
        "value": "",
        "previewImage": "t84_1.jpg"
      }
    },
    "ui": [
      {
        "Properties": [
          {
            "type": "TextField",
            "name": "Top Line"
          },
          {
            "type": "TextField",
            "name": "Bottom Line"
          }
        ]
      }
    ]
  },
  "91": {
    "templateId": 91,
    "textFields": {
      "Top Line": {
        "value": "",
        "previewImage": "t91_1.jpg"
      },
      "Bottom Line": {
        "value": "",
        "previewImage": "t91_1.jpg"
      }
    },
    "ui": [
      {
        "Properties": [
          {
            "type": "TextField",
            "name": "Top Line"
          },
          {
            "type": "TextField",
            "name": "Bottom Line"
          }
        ]
      }
    ]
  },
  "79": {
    "templateId": 79,
    "dropDowns": {
      "Icon Style": {
        "default": "MyOwnCharacter",
        "options": [
          {
            "name": "My Own Character",
            "value": "MyOwnCharacter",
            "previewImage": "79_MyOwnCharacter.jpg"
          },
          {
            "name": "Android", 
            "value": "android", 
            "previewImage": "79_android.jpg"
          },
          {
            "name": "Apple", 
            "value": "apple", 
            "previewImage": "79_apple.jpg"
          },
          {
            "name": "Basketball", 
            "value": "basketball", 
            "previewImage": "79_basketball.jpg"
          },
          {
            "name": "Broadcast", 
            "value": "broadcast", 
            "previewImage": "79_broadcast.jpg"
          },
          {
            "name": "Broadcast1", 
            "value": "broadcast1", 
            "previewImage": "79_broadcast1.jpg"
          },
          {
            "name": "Buicong", 
            "value": "buicong", 
            "previewImage": "79_buicong.jpg"
          },
          {
            "name": "Bullhorn", 
            "value": "bullhorn", 
            "previewImage": "79_bullhorn.jpg"
          },
          {
            "name": "Business", 
            "value": "business", 
            "previewImage": "79_business.jpg"
          },
          {
            "name": "Camera", 
            "value": "camera", 
            "previewImage": "79_camera.jpg"
          },
          {
            "name": "Car", 
            "value": "car", 
            "previewImage": "79_car.jpg"
          },
          {
            "name": "Cart", 
            "value": "cart", 
            "previewImage": "79_cart.jpg"
          },
          {
            "name": "Chart", 
            "value": "chart", 
            "previewImage": "79_chart.jpg"
          },
          {
            "name": "Clock", 
            "value": "clock", 
            "previewImage": "79_clock.jpg"
          },
          {
            "name": "Comment", 
            "value": "comment", 
            "previewImage": "79_comment.jpg"
          },
          {
            "name": "Crosshairs", 
            "value": "crosshairs", 
            "previewImage": "79_crosshairs.jpg"
          },
          {
            "name": "Crown", 
            "value": "crown", 
            "previewImage": "79_crown.jpg"
          },
          {
            "name": "Earth", 
            "value": "earth", 
            "previewImage": "79_earth.jpg"
          },
          {
            "name": "Energy", 
            "value": "energy", 
            "previewImage": "79_energy.jpg"
          },
          {
            "name": "Eye", 
            "value": "eye", 
            "previewImage": "79_eye.jpg"
          },
          {
            "name": "Facebook", 
            "value": "facebook", 
            "previewImage": "79_facebook.jpg"
          },
          {
            "name": "Film", 
            "value": "film", 
            "previewImage": "79_film.jpg"
          },
          {
            "name": "Fire", 
            "value": "fire", 
            "previewImage": "79_fire.jpg"
          },
          {
            "name": "Flag", 
            "value": "flag", 
            "previewImage": "79_flag.jpg"
          },
          {
            "name": "Gauge", 
            "value": "gauge", 
            "previewImage": "79_gauge.jpg"
          },
          {
            "name": "Gear", 
            "value": "gear", 
            "previewImage": "79_gear.jpg"
          },
          {
            "name": "Heart", 
            "value": "heart", 
            "previewImage": "79_heart.jpg"
          },
          {
            "name": "House", 
            "value": "house", 
            "previewImage": "79_house.jpg"
          },
          {
            "name": "Key", 
            "value": "key", 
            "previewImage": "79_key.jpg"
          },
          {
            "name": "Legal", 
            "value": "legal", 
            "previewImage": "79_legal.jpg"
          },
          {
            "name": "Lightbulb", 
            "value": "lightbulb", 
            "previewImage": "79_lightbulb.jpg"
          },
          {
            "name": "Lock", 
            "value": "lock", 
            "previewImage": "79_lock.jpg"
          },
          {
            "name": "Magic", 
            "value": "magic", 
            "previewImage": "79_magic.jpg"
          },
          {
            "name": "Mobile", 
            "value": "mobile", 
            "previewImage": "79_mobile.jpg"
          },
          {
            "name": "Movie", 
            "value": "movie", 
            "previewImage": "79_movie.jpg"
          },
          {
            "name": "On Button", 
            "value": "onbutton", 
            "previewImage": "79_onbutton.jpg"
          },
          {
            "name": "Plane", 
            "value": "plane", 
            "previewImage": "79_plane.jpg"
          },
          {
            "name": "Rocket", 
            "value": "rocket", 
            "previewImage": "79_rocket.jpg"
          },
          {
            "name": "Scull & Bones", 
            "value": "scullbones", 
            "previewImage": "79_scullbones.jpg"
          },
          {
            "name": "Search", 
            "value": "search", 
            "previewImage": "79_search.jpg"
          },
          {
            "name": "Smiley", 
            "value": "smiley", 
            "previewImage": "79_smiley.jpg"
          },
          {
            "name": "Speaker", 
            "value": "speaker", 
            "previewImage": "79_speaker.jpg"
          },
          {
            "name": "Star", 
            "value": "star", 
            "previewImage": "79_star.jpg"
          },
          {
            "name": "Target", 
            "value": "target", 
            "previewImage": "79_target.jpg"
          },
          {
            "name": "Thumbs Up", 
            "value": "thumbsup", 
            "previewImage": "79_thumbsup.jpg"
          },
          {
            "name": "Tools", 
            "value": "tools", 
            "previewImage": "79_tools.jpg"
          },
          {
            "name": "Twitter", 
            "value": "twitter", 
            "previewImage": "79_twitter.jpg"
          },
          {
            "name": "Xbox", 
            "value": "xbox", 
            "previewImage": "79_xbox.jpg"
          },
          {
            "name": "YouTube", 
            "value": "youtube", 
            "previewImage": "79_youtube.jpg"
          },
        ]
      },
      "Background": {
        "default": "paper",
        "options": [
          {
            "name": "Brick", 
            "value": "brick", 
            "previewImage": "79_brick.jpg"
          },
          {
            "name": "Camo", 
            "value": "camo", 
            "previewImage": "79_camo.jpg"
          },
          {
            "name": "Concrete", 
            "value": "concrete", 
            "previewImage": "79_concrete.jpg"
          },
          {
            "name": "Grass", 
            "value": "grass", 
            "previewImage": "79_grass.jpg"
          },
          {
            "name": "Ground", 
            "value": "ground", 
            "previewImage": "79_ground.jpg"
          },
          {
            "name": "Ice", 
            "value": "ice", 
            "previewImage": "79_ice.jpg"
          },
          {
            "name": "Marble", 
            "value": "marble", 
            "previewImage": "79_marble.jpg"
          },
          {
            "name": "Metal", 
            "value": "metal", 
            "previewImage": "79_metal.jpg"
          },
          {
            "name": "Paper", 
            "value": "paper",
            "previewImage": "79_paper.jpg"
          },
          {
            "name": "Wood", 
            "value": "wood", 
            "previewImage": "79_wood.jpg"
          }
        ]
      },
      "Icon Color": {
        "default": "icon_black",
        "options": [
          {
            "name": "Black",  
            "value": "icon_black", 
            "previewImage": "79_icon_black.jpg"
          },
          {
            "name": "Blue",  
            "value": "icon_blue", 
            "previewImage": "79_icon_blue.jpg"
          },
          {
            "name": "Green",  
            "value": "icon_green", 
            "previewImage": "79_icon_green.jpg"
          },
          {
            "name": "Orange",  
            "value": "icon_orange", 
            "previewImage": "79_icon_orange.jpg"
          },
          {
            "name": "Pink",  
            "value": "icon_pink", 
            "previewImage": "79_icon_pink.jpg"
          },
          {
            "name": "Puprle",  
            "value": "icon_puprle", 
            "previewImage": "79_icon_puprle.jpg"
          },
          {
            "name": "Red",  
            "value": "icon_red", 
            "previewImage": "79_icon_red.jpg"
          },
          {
            "name": "Silver",  
            "value": "icon_silver", 
            "previewImage": "79_icon_silver.jpg"
          },
          {
            "name": "Teal",  
            "value": "icon_teal", 
            "previewImage": "79_icon_teal.jpg"
          },
          {
            "name": "Yellow",  
            "value": "icon_yellow", 
            "previewImage": "79_icon_yellow.jpg"
          }
        ]
      },
      "Main Text Color": {
        "default": "main_black",
        "options": [
          {
            "name": "Black",
            "value": "main_black",
            "previewImage": "79_main_black.jpg"
          },
          {
            "name": "Blue",
            "value": "main_blue",
            "previewImage": "79_main_blue.jpg"
          },
          {
            "name": "Green",
            "value": "main_green",
            "previewImage": "79_main_green.jpg"
          },
          {
            "name": "Orange",
            "value": "main_orange",
            "previewImage": "79_main_orange.jpg"
          },
          {
            "name": "Pink",
            "value": "main_pink",
            "previewImage": "79_main_pink.jpg"
          },
          {
            "name": "Puprle",
            "value": "main_puprle",
            "previewImage": "79_main_puprle.jpg"
          },
          {
            "name": "Red",
            "value": "main_red",
            "previewImage": "79_main_red.jpg"
          },
          {
            "name": "Silver",
            "value": "main_silver",
            "previewImage": "79_main_silver.jpg"
          },
          {
            "name": "Teal",
            "value": "main_teal",
            "previewImage": "79_main_teal.jpg"
          },
          {
            "name": "Yellow",
            "value": "main_yellow",
            "previewImage": "79_main_yellow.jpg"
          }
        ]
      },
      "Subtitle Text Color": {
        "default": "sub_black",
        "options": [
          {
            "name": "Black",
            "value": "sub_black",
            "previewImage": "79_sub_black.jpg"
          },
          {
            "name": "White",
            "value": "sub_white",
            "previewImage": "79_sub_white.jpg"
          },
          {
            "name": "Orange",
            "value": "sub_orange",
            "previewImage": "79_sub_orange.jpg"
          },
          {
            "name": "Teal",
            "value": "sub_teal",
            "previewImage": "79_sub_teal.jpg"
          },
          {
            "name": "Yellow",
            "value": "sub_yellow",
            "previewImage": "79_sub_yellow.jpg"
          }
        ]
      }
    },
    "textFields": {
      "Icon Character - type in only one (optional)": {
        "value": "",
        "previewImage": "79_letter.jpg"
      },
      "Text Left of Icon": {
        "value": "",
        "previewImage": "79_letter.jpg"
      },
      "Text Right of Icon": {
        "value": "",
        "previewImage": "79_letter.jpg"
      },
      "Subtitle": {
        "value": "",
        "previewImage": "79_letter.jpg"
      }
    },
    "ui": [
      {
        "Customize Icon": [
          {
            "type": "DropDown",
            "name": "Icon Style"
          },
          {
            "type": "TextField",
            "name": "Icon Character - type in only one (optional)"
          }
        ]
      },
      {
        "Customize Look": [
          {
            "type": "DropDown",
            "name": "Background"
          },
          {
            "type": "DropDown",
            "name": "Icon Color"
          },
          {
            "type": "DropDown",
            "name": "Main Text Color"
          },
          {
            "type": "DropDown",
            "name": "Subtitle Text Color"
          }
        ]
      },
      {
        "Customize Text": [
          {
            "type": "TextField",
            "name": "Text Left of Icon"
          },
          {
            "type": "TextField",
            "name": "Text Right of Icon"
          },
          {
            "type": "TextField",
            "name": "Subtitle"
          }
        ]
      }
    ]
  }
}

export default fakeTemplateInfo;