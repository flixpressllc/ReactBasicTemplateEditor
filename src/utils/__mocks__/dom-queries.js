'use strict';
// export function find(selector) {
//   return $(selector);
// }

var __mockElements = {};

export function getElementById (id) {
  return __mockElements[id];
}

export function __setMockElement(id, object) {
  __mockElements[id] = object;
}
