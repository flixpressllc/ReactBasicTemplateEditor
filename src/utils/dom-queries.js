export function find(selector) {
  return $(selector).get();
}

export function getElementById (id) {
  return document.getElementById(id);
}
