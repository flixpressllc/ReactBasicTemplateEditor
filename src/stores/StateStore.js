import { EventEmitter } from 'events';
import dispatcher from '../actions/dispatcher';
import { clone, traverseObject, toType } from 'happy-helpers';
import { isDevelopment } from './app-settings';

const ALLOWED_STATES = {
  templateType: 'string',
  templateId: 'number',
  caughtErrors: 'array',
  username: 'string',
  usernames: 'string'
}

class StateStore extends EventEmitter {
  constructor () {
    super();
    this.state = {
      caughtErrors: []
    };
  }

  getState (stringOrArrayOfKeys) {
    if (stringOrArrayOfKeys === undefined) return clone(this.state);
    if (toType(stringOrArrayOfKeys) === 'array') {
      return this.getStateByArrayOfKeys(stringOrArrayOfKeys);
    }
    if (toType(stringOrArrayOfKeys) === 'string') {
      return this.getStateByKey(stringOrArrayOfKeys);
    }
  }

  // Returns a value only
  getStateByKey (key) {
    if (ALLOWED_STATES[key] === undefined) {
      throw new Error(`The desired state property "${key}" from StateStore is not allowed.`);
    }
    if (['object', 'array'].indexOf(ALLOWED_STATES[key]) > -1) {
      return clone(this.state[key]);
    }
    return this.state[key];
  }

  // Returns an object of key: value pairs
  getStateByArrayOfKeys (arrayOfKeys) {
    let objects = arrayOfKeys.map(key => {
      return {[key]: this.getStateByKey(key)} // ES2015 computed property names
    });
    return Object.assign(...objects);
  }

  setState (obj) {
    this.statesAreAllowed(obj);
    let newState = Object.assign({}, this.state, obj);
    if (!this.statesAreEqual(newState)) {
      this.state = newState;
      this.emit('STATE_UPDATED');
    }
  }

  statesAreAllowed (obj) {
    traverseObject(obj, (k, v) => {
      if (ALLOWED_STATES[k] === undefined) {
        throw new Error(`There is no entry for "${k}" in StateStore's allowed states.`);
      }
      if (ALLOWED_STATES[k] !== toType(v)) {
        throw new Error(`StateStore was expecting the type of ${k} to be ${ALLOWED_STATES[k]}. ${toType(v)} was given.`)
      }
    })
  }

  stateExists (key) {
    return this.state[key] !== undefined;
  }

  stateValueEquals (key, newValue) {
    if (!this.stateExists(key)) return false;
    let stateValue = this.state[key];
    let theType = toType(stateValue);

    if (theType !== toType(newValue)) return false;
    if (['array','object'].indexOf(theType) > -1) {
      return JSON.stringify(newValue) === JSON.stringify(stateValue);
    }
    return newValue === stateValue;
  }

  statesAreEqual (newState) {
    let equal = true;
    traverseObject(newState, (k, v) => {
      if (!this.stateValueEquals(k, v)) {
        equal = false;
      }
    })
    return equal;
  }

  handleActions(action) {
    switch(action.type) {
      case 'SET_STATE':
        this.setState(action.state);
      break;
      default: break;
    }
  }
}

const stateStore = new StateStore();
dispatcher.register(stateStore.handleActions.bind(stateStore))

if (isDevelopment()) {
  window.StateStore = stateStore;
}

export default stateStore;
