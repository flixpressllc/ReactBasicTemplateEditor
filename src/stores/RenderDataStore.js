import { EventEmitter } from 'events';
import dispatcher from './dispatcher';
import * as Containers from '../utils/globalContainerConcerns';

import { clone, isObject } from '../utils/helper-functions';

class RenderDataStore extends EventEmitter {
  constructor () {
    super();
    this.state = {
      receivedInitial: false
    }
  }
  changeContainer(dataType, fieldName, newData) {
    let containerName = Containers.getContainerNameFor(dataType);
    Object.assign(this.containers[containerName][fieldName], newData);
    this.emit('change', {containerName, dataType, fieldName});
  }
  
  addMissingContainers() {
    if (!isObject(this.containers)) {
      this.containers = {};
    }
    Containers.getContainerNames().map(name => {
      if (!this.containers.hasOwnProperty(name)) {
        this.containers[name] = {};
      }
    })
  }

  getAll() {
    if ( ! this.state.receivedInitial) this.addMissingContainers();
    return this.containers ? clone(this.containers) : {};
  }
  
  handleActions(action) {
    switch(action.type) {
      case 'CHANGE_CONTAINER':
        this.changeContainer(action.dataType, action.fieldName, action.newData);
      break;
      case 'RECEIVE_INITIAL_CONTAINER_VALUES':
        this.state.receivedInitial = true;
        this.containers = action.containers;
        this.emit('change');
      break;
    }
  }
}

const renderDataStore = new RenderDataStore;
dispatcher.register(renderDataStore.handleActions.bind(renderDataStore))
window.renderDataStore = renderDataStore;
export default renderDataStore;
