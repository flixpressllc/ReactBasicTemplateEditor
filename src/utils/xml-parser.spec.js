require('jxon');
import Parser from './xml-parser';
import jsb from 'js-beautify';
import { XML_CONTAINER_ID } from '../stores/app-settings';

jest.mock('./dom-queries');

let pretty = (obj) => jsb.html(obj);

const startingTextOnlyXml = `<?xml version="1.0" encoding="utf-16"?>
<OrderRequestOfTextOnlyRndTemplate xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
<ResolutionId>0</ResolutionId>
<IsPreview>false</IsPreview>
<ResolutionOptions>
  <ListItemViewModel>
    <Id>5</Id>
    <Name>720p</Name>
  </ListItemViewModel>
  <ListItemViewModel>
    <Id>3</Id>
    <Name>1080p</Name>
  </ListItemViewModel>
  <ListItemViewModel>
    <Id>4</Id>
    <Name>4K</Name>
  </ListItemViewModel>
</ResolutionOptions>
</OrderRequestOfTextOnlyRndTemplate>`;

const mockOrderObjForTextOnly79 = {
  "ui": [
    {
      "Customize Icon": [
        {
          "type": "DropDown",
          "name": "Icon Style",
          "value": "camera"
        },
        {
          "type": "TextField",
          "name": "Icon Character - type in only one (optional)",
          "value": "q"
        }
      ]
    },
    {
      "Customize Look": [
        {
          "type": "DropDown",
          "name": "Background",
          "value": "camo"
        },
        {
          "type": "DropDown",
          "name": "Icon Color",
          "value": "icon_pink"
        },
        {
          "type": "DropDown",
          "name": "Main Text Color",
          "value": "main_orange"
        },
        {
          "type": "DropDown",
          "name": "Subtitle Text Color",
          "value": "sub_white"
        }
      ]
    },
    {
      "Customize Text": [
        {
          "type": "TextField",
          "name": "Text Left of Icon",
          "value": "Left"
        },
        {
          "type": "TextField",
          "name": "Text Right of Icon",
          "value": "Right"
        },
        {
          "type": "TextField",
          "name": "Subtitle",
          "value": "Bottom"
        }
      ]
    }
  ],
  "isPreview": true,
  "audioInfo": {
    "audioType": "StockAudio",
    "audioUrl": "https://fpsound.s3.amazonaws.com/13.mp3",
    "id": 13,
    "length": 3,
    "name": "Bunny Garden"
  },
  "resolutionId": 5
};

describe('Parser', () => {
  describe('getReactStartingData', () => {
    it('returns the expected object for a new TextOnly', () => {
      require('./dom-queries').__setMockElement(XML_CONTAINER_ID, {value: startingTextOnlyXml})
      expect(Parser.getReactStartingData()).toMatchSnapshot();
    });
  });
  describe('updateXmlForOrder', () => {
    it('updates the container with the expected values for new TextOnly', () => {
      let dom = require('./dom-queries');
      dom.__setMockElement(XML_CONTAINER_ID, {value: startingTextOnlyXml});

      Parser.updateXmlForOrder(mockOrderObjForTextOnly79);

      expect(pretty(dom.getElementById(XML_CONTAINER_ID).value)).toMatchSnapshot();
    });
  });
});
