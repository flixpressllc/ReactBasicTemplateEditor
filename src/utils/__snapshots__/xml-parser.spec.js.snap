exports[`Parser getReactStartingData returns the expected object for a new TextOnly 1`] = `
Object {
  "isPreview": false,
  "resolutionId": 5,
  "resolutions": Array [
    Object {
      "id": 5,
      "name": "720p",
    },
    Object {
      "id": 3,
      "name": "1080p",
    },
    Object {
      "id": 4,
      "name": "4K",
    },
  ],
}
`;

exports[`Parser updateXmlForOrder updates the container with the expected values for new TextOnly 1`] = `
"<?xml version="1.0" encoding="utf-16"?>
<OrderRequestOfTextOnlyRndTemplate xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
    <ResolutionId>5</ResolutionId>
    <RenderedData>
        <Specs name="Specs" val="">
            <SpCx>
                <CSp name="Customize Icon" val="CD|Customize Icon|">
                    <SpCx>
                        <Sp name="Icon Style" val="camera" />
                        <Sp name="Icon Character - type in only one (optional)" val="q" />
                    </SpCx>
                </CSp>
                <CSp name="Customize Look" val="CD|Customize Look|">
                    <SpCx>
                        <Sp name="Background" val="camo" />
                        <Sp name="Icon Color" val="icon_pink" />
                        <Sp name="Main Text Color" val="main_orange" />
                        <Sp name="Subtitle Text Color" val="sub_white" />
                    </SpCx>
                </CSp>
                <CSp name="Customize Text" val="CD|Customize Text|">
                    <SpCx>
                        <Sp name="Text Left of Icon" val="Left" />
                        <Sp name="Text Right of Icon" val="Right" />
                        <Sp name="Subtitle" val="Bottom" />
                    </SpCx>
                </CSp>
            </SpCx>
        </Specs>
        <AudioInfo>
            <AudioType>StockAudio</AudioType>
            <AudioUrl>https://fpsound.s3.amazonaws.com/13.mp3</AudioUrl>
            <Id>13</Id>
            <Length>3</Length>
            <Name>Bunny Garden</Name>
        </AudioInfo>
    </RenderedData>
    <IsPreview>true</IsPreview>
</OrderRequestOfTextOnlyRndTemplate>"
`;
